src/ZODB/DB.py

```diff
@@ -646,15 +646,15 @@ def close(self):
         is closed, so they stop behaving usefully.  Perhaps close()
         should also close all the Connections.
         """
-        noop = lambda *a: None
         self.close = noop
 
         @self._connectionMap
-        def _(c):
-            if c.transaction_manager is not None:
-                c.transaction_manager.abort()
-            c.afterCompletion = c.newTransaction = c.close = noop
-            c._release_resources()
+        def _(conn):
+            if conn.transaction_manager is not None:
+                for c in six.itervalues(conn.connections):
+                    c.explicit_transactions = True
+                conn.transaction_manager.abort()
```

Comment:
The 2 other big changes:
- Make sure that the real `newTransaction` is not called when aborting an existing transaction. This was not the case if `explicit_transactions=0`.
- As written above, Connections of closed DB could remain registered, and when it happened, multi-db was probably broken because only the primary connections were ignored.

Setting `explicit_transactions=1` is also simpler than patching newTransaction.